### DISCUSSION TITLE
Store file after restart dev

### DISCUSSION THREAD

**Discussion by quangthienit1994** [adonisjs/core#1782] [Help] [answered]
Upload module working very perfect, but I don't know how to stop Adonis auto delete all file uploaded to public folder after restart dev environment. :(

**Comment by thetutlage** [comment, marked as answer]
The uploaded file shouldn't be stored inside the `public` folder. You must store inside a directory outside of the `build` folder and then create a route to serve the files. Roughly the code should look as follows:

```ts
await file.move(Application.tmpPath('uploads'))
```

and then serve them as follows:

```ts
Route.get('uploads/:filename', ({ response, params }) => {
  return response.download(Application.tmpPath('uploads', params.filename))
})
```

The great thing about custom route is that it gives you the freedom to have custom logic when serving files to the user

**Reply by quangthienit1994** [reply]
I don't want download the file, that only images of product showing on the website. It's still working with response.download ? I only want use all file uploaded like https://preview.adonisjs.com/guides/http/static-files :))

**Reply by thetutlage** [reply]
The static files are for resources that are fixed and static. For example: Css, Javascript files or images that are not uploaded by the user.

Follow the above shared answer

**Reply by sasha172017** [reply]
how can i dowload file which is in other folders, for example "uploads/image/news/image.png"?
